hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationFuture.java

```diff
@@ -110,11 +112,16 @@ protected Object resolve(Object unresolved) {
         Object value = unresolved;
         if (deserialize && value instanceof Data) {
             value = invocation.context.serializationService.toObject(value);
+
             if (value == null) {
                 return null;
             }
         }
 
+        if (invocation.shouldFailOnOperationBackupTimeout() && (value instanceof MemberLeftException)) {
+            value = new OperationBackupAckTimeoutException();
```

Comment:
This is a different cause of the exception; can you add as much context as possible to the exception message?